### PR TITLE
Adding the files from the Roadster branch to do a pull request

### LINKED ISSUE #13: Start road mapping tool

### LINKED ISSUE #30: Add Splines of Curves
Add functionality for creating curves built up of several curves, splines.

### LINKED ISSUE #31: Roads following curves
Create a 3D model which follows the direction of curves.

### PR DESCRIPTION
**Pull request by Keywarn**
Finished the roadster tool for creating road meshes. Has to create a separate branch to be able to pull only the relevant files and avoid a messy merge. For full history look at the branch `tool-roadster`.

Resolves #13
Resolves #30
Resolves #31

### CODE REVIEW

File: ProjectFiles/Assets/Tools/Roadster/Editor/BezierSplineInspector copy.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+//                Vector3 lineEnd = curve.GetPoint(i / (float)lineSteps);
+//                Handles.DrawLine(lineStart, lineEnd);
+//                lineStart = lineEnd;
+//            }
+            Handles.DrawBezier(p0, p3, p1, p2, Color.white, null, 2f);
+            p0 = p3;
+        }
+    }    
+    
+    private static Color[] modeColors = {
```
Comment (RileyEv) [inline ProjectFiles/Assets/Tools/Roadster/Editor/BezierSplineInspector copy.cs]: top of the file?

**Review by RileyEv — CHANGES_REQUESTED** [review]
- Can we put it in a namespace?
- `BezierSplineInspector copy.cs` ?
- For some reason the multiplayer submodule seems to have creeped back in...

File: ProjectFiles/Assets/Tools/Roadster/Editor/BezierSplineInspector copy.cs (hunk 1 above)
Comment (Keywarn) [reply ProjectFiles/Assets/Tools/Roadster/Editor/BezierSplineInspector copy.cs]: Moving it, not sure why I put it there...

**Review by Keywarn — COMMENTED** [review]
- I guess I can put in a namespace but not sure I see the benefit?
- Renamed the file, not entirely sure why it had copy on the end of it.
- Not sure why the multiplayer module was back, I thought I had deleted it, should be gone now.
- Moved the mode colours to the top of the files

**Review by RileyEv — APPROVED** [review]

**Pull request merged**